### PR TITLE
go/oasis-test-runner: List all available scenarios

### PR DESCRIPTION
**Pull request by tjanez**
List all available scenarios if command was invoked with a particular scenario that was not found.

For example, running `.buildkite/scripts/test_e2e.sh --test storage-sync` where `storage-sync` scenario doesn't exist, now returns:

```
+ WORKDIR=/home/tadej/Oasis/oasis-core
+ runtime_target=default
+ [[ '' == \i\n\t\e\l\-\s\g\x ]]
+ [[ '' != '' ]]
+ node_binary=/home/tadej/Oasis/oasis-core/go/oasis-node/oasis-node
+ test_runner_binary=/home/tadej/Oasis/oasis-core/go/oasis-test-runner/oasis-test-runner
+ [[ '' != '' ]]
+ ias_mock=true
+ set +x
+ /home/tadej/Oasis/oasis-core/go/oasis-test-runner/oasis-test-runner --basedir.no_cleanup --e2e.node.binary /home/tadej/Oasis/oasis-core/go/oasis-node/oasis-node --e2e/runtime.client.binary_dir /home/tadej/Oasis/oasis-core/target/default/debug --e2e/runtime.runtime.binary_dir /home/tadej/Oasis/oasis-core/target/default/debug --e2e/runtime.runtime.loader /home/tadej/Oasis/oasis-core/target/default/debug/oasis-core-runtime-loader --e2e/runtime.tee_hardware '' --e2e/runtime.ias.mock=true --remote-signer.binary /home/tadej/Oasis/oasis-core/go/oasis-remote-signer/oasis-remote-signer --log.level info --test storage-sync
level=error module=test-runner caller=root.go:279 ts=2020-06-24T08:07:30.560481983Z msg="unknown scenario" scenario=storage-sync
Error: root: unknown scenario: storage-sync
Available scenarios:
e2e/identity-cli
e2e/runtime/byzantine/executor-honest
e2e/runtime/byzantine/executor-straggler
e2e/runtime/byzantine/executor-wrong
e2e/runtime/byzantine/merge-honest
e2e/runtime/byzantine/merge-straggler
e2e/runtime/byzantine/merge-wrong
e2e/runtime/debond
e2e/runtime/dump-restore
e2e/runtime/early-query
e2e/runtime/gas-fees/runtimes
e2e/runtime/gas-fees/staking
e2e/runtime/gas-fees/staking-dump-restore
e2e/runtime/halt-restore
e2e/runtime/keymanager-replication
e2e/runtime/keymanager-restart
e2e/runtime/keymanager-upgrade
e2e/runtime/late-start
e2e/runtime/multiple-runtimes
e2e/runtime/node-shutdown
e2e/runtime/node-upgrade
e2e/runtime/node-upgrade-cancel
e2e/runtime/registry-cli
e2e/runtime/runtime
e2e/runtime/runtime-dynamic
e2e/runtime/runtime-encryption
e2e/runtime/runtime-prune
e2e/runtime/sentry
e2e/runtime/sentry-encryption
e2e/runtime/stake-cli
e2e/runtime/storage-sync
e2e/runtime/txsource-multi
e2e/runtime/txsource-multi-short
remote-signer/basic
```

### CODE REVIEW

File: go/oasis-test-runner/cmd/common/common.go [hunk 1]
```diff
@@ -33,6 +34,14 @@ func GetScenarios() map[string]scenario.Scenario {
 	return scenarios
 }
 
+func GetScenarioNames() (names []string) {
```
Comment (kostko) [inline go/oasis-test-runner/cmd/common/common.go:37]: Documentation comment.

File: go/oasis-test-runner/cmd/root.go [hunk 2]
```diff
@@ -75,7 +75,7 @@ func RootCmd() *cobra.Command {
 // Execute spawns the main entry point after handing the config file.
 func Execute() {
 	if err := rootCmd.Execute(); err != nil {
-		nodeCommon.EarlyLogAndExit(err)
```
Comment (kostko) [inline go/oasis-test-runner/cmd/root.go:78]: Why this change?

Comment (tjanez) [reply go/oasis-test-runner/cmd/root.go:78]: The `nodeCommon.EarlyLogAndExit()` prints the error to stderr:
https://github.com/oasisprotocol/oasis-core/blob/ffa545427af15995184633455177ea5b1ad4e8af/go/oasis-node/cmd/common/common.go#L79-L86

which results in the same error message being outputted twice.

For example:
```
$ .buildkite/scripts/test_e2e.sh --test storage-sync
+ WORKDIR=/home/tadej/Oasis/oasis-core
+ runtime_target=default
+ [[ '' == \i\n\t\e\l\-\s\g\x ]]
+ [[ '' != '' ]]
+ node_binary=/home/tadej/Oasis/oasis-core/go/oasis-node/oasis-node
+ test_runner_binary=/home/tadej/Oasis/oasis-core/go/oasis-test-runner/oasis-test-runner
+ [[ '' != '' ]]
+ ias_mock=true
+ set +x
+ /home/tadej/Oasis/oasis-core/go/oasis-test-runner/oasis-test-runner --basedir.no_cleanup --e2e.node.binary /home/tadej/Oasis/oasis-core/go/oasis-node/oasis-node --e2e/runtime.client.binary_dir /home/tadej/Oasis/oasis-core/target/default/debug --e2e/runtime.runtime.binary_dir /home/tadej/Oasis/oasis-core/target/default/debug --e2e/runtime.runtime.loader /home/tadej/Oasis/oasis-core/target/default/debug/oasis-core-runtime-loader --e2e/runtime.tee_hardware '' --e2e/runtime.ias.mock=true --remote-signer.binary /home/tadej/Oasis/oasis-core/go/oasis-remote-signer/oasis-remote-signer --log.level info --test storage-sync
level=error module=test-runner caller=root.go:279 ts=2020-06-24T08:24:40.945179021Z msg="unknown scenario" scenario=storage-sync
Error: root: unknown scenario: storage-sync
Available scenarios:
e2e/identity-cli
e2e/runtime/byzantine/executor-honest
e2e/runtime/byzantine/executor-straggler
e2e/runtime/byzantine/executor-wrong
e2e/runtime/byzantine/merge-honest
e2e/runtime/byzantine/merge-straggler
e2e/runtime/byzantine/merge-wrong
e2e/runtime/debond
e2e/runtime/dump-restore
e2e/runtime/early-query
e2e/runtime/gas-fees/runtimes
e2e/runtime/gas-fees/staking
e2e/runtime/gas-fees/staking-dump-restore
e2e/runtime/halt-restore
e2e/runtime/keymanager-replication
e2e/runtime/keymanager-restart
e2e/runtime/keymanager-upgrade
e2e/runtime/late-start
e2e/runtime/multiple-runtimes
e2e/runtime/node-shutdown
e2e/runtime/node-upgrade
e2e/runtime/node-upgrade-cancel
e2e/runtime/registry-cli
e2e/runtime/runtime
e2e/runtime/runtime-dynamic
e2e/runtime/runtime-encryption
e2e/runtime/runtime-prune
e2e/runtime/sentry
e2e/runtime/sentry-encryption
e2e/runtime/stake-cli
e2e/runtime/storage-sync
e2e/runtime/txsource-multi
e2e/runtime/txsource-multi-short
remote-signer/basic
root: unknown scenario: storage-sync
Available scenarios:
e2e/identity-cli
e2e/runtime/byzantine/executor-honest
e2e/runtime/byzantine/executor-straggler
e2e/runtime/byzantine/executor-wrong
e2e/runtime/byzantine/merge-honest
e2e/runtime/byzantine/merge-straggler
e2e/runtime/byzantine/merge-wrong
e2e/runtime/debond
e2e/runtime/dump-restore
e2e/runtime/early-query
e2e/runtime/gas-fees/runtimes
e2e/runtime/gas-fees/staking
e2e/runtime/gas-fees/staking-dump-restore
e2e/runtime/halt-restore
e2e/runtime/keymanager-replication
e2e/runtime/keymanager-restart
e2e/runtime/keymanager-upgrade
e2e/runtime/late-start
e2e/runtime/multiple-runtimes
e2e/runtime/node-shutdown
e2e/runtime/node-upgrade
e2e/runtime/node-upgrade-cancel
e2e/runtime/registry-cli
e2e/runtime/runtime
e2e/runtime/runtime-dynamic
e2e/runtime/runtime-encryption
e2e/runtime/runtime-prune
e2e/runtime/sentry
e2e/runtime/sentry-encryption
e2e/runtime/stake-cli
e2e/runtime/storage-sync
e2e/runtime/txsource-multi
e2e/runtime/txsource-multi-short
remote-signer/basic
```

Comment (kostko) [reply go/oasis-test-runner/cmd/root.go:78]: Who prints the first error message, does cobra do that internally?

Comment (tjanez) [reply go/oasis-test-runner/cmd/root.go:78]: Yes, Cobra does that in [`ExecuteC()`](https://github.com/spf13/cobra/blob/a684a6d7f5e37385d954dd3b5a14fc6912c6ab9d/command.go#L959-L963):

```go
// If root command has SilentErrors flagged,
// all subcommands should respect it
if !cmd.SilenceErrors && !c.SilenceErrors {
	c.Println("Error:", err.Error())
}
```

Comment (kostko) [reply go/oasis-test-runner/cmd/root.go:78]: I like our helper better as it prints to stderr while cobra seems to print to stdout? Could we instead set `SilenceErrors` (maybe this needs to be done in other places as well as we use `EarlyLogAndExit` everywhere)?

File: go/oasis-test-runner/cmd/common/common.go (hunk 1 above)
Comment (tjanez) [reply go/oasis-test-runner/cmd/common/common.go:38]: Done.

File: go/oasis-test-runner/cmd/root.go (hunk 2 above)
Comment (tjanez) [reply go/oasis-test-runner/cmd/root.go:78]: > I like our helper better as it prints to stderr while cobra seems to print to stdout?

[Cobra's `Command.Println()`](https://pkg.go.dev/github.com/spf13/cobra?tab=doc#Command.Println) defaults to printing to stderr.

Comment (kostko) [reply go/oasis-test-runner/cmd/root.go:78]: Why do we have this `EarlyLogAndExit` at all then?

Comment (tjanez) [reply go/oasis-test-runner/cmd/root.go:78]: I guess it is still useful for cases where we want to exit and print an error before we execute Cobra's `Execute()`.

I found two such cases in `go/oasis-test-runner`:
- https://github.com/oasisprotocol/oasis-core/blob/ffa545427af15995184633455177ea5b1ad4e8af/go/oasis-test-runner/test-runner.go#L26
- https://github.com/oasisprotocol/oasis-core/blob/f5f906b3c076290d20ea6cbd298ce548c2000513/go/oasis-test-runner/cmd/root.go#L550

I've removed other uses in `go/oasis-test-runner`.

For other parts of the code base, I can file an issue to systematically check that.

Comment (tjanez) [reply go/oasis-test-runner/cmd/root.go:78]: > For other parts of the code base, I can file an issue to systematically check that.

Done: #3048.
